roscala/src/main/scala/coop/rchain/rosette/prim/Tuple.scala

```diff
@@ -0,0 +1,37 @@
+package coop.rchain.rosette.prim
+
+import coop.rchain.rosette.{
+  Ctxt,
+  Fixnum => RFixnum,
+  Ob,
+  RblBool,
+  Tuple => RTuple
+}
+import coop.rchain.rosette.macros.{checkArgumentMismatch, checkTypeMismatch}
+import coop.rchain.rosette.prim.Prim._
+
+object Tuple {
+  object tplCons extends Prim {
+    override val name: String = "tuple-cons"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    //@checkTypeMismatch[RTuple]    arg0 can be any Ob
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RTuple] = {
+      val elem = ctxt.argvec.elem
+      checkRTuple(1, elem).map { tup => // arg1 must be a Tuple
+        RTuple.cons(elem(0), tup)
+      }
+    }
+
+  }
+
+  private def checkRTuple(n: Int, elem: Seq[Ob]): Either[PrimError, RTuple] =
+    if (!elem(n).isInstanceOf[RTuple]) {
+      Left(TypeMismatch(n, RTuple.getClass().getName()))
```

Comment:
It's convention to use `RTuple.getClass.getName` here since `getClass` and `getName` have no side effects. At least that's what IntelliJ is telling me ;)

Reply:
Sorry, my C/C++ roots are showing...